pallets/pallet-deposits/src/deposit_change_handler.rs

```diff
@@ -0,0 +1,39 @@
+use frame_system::pallet_prelude::OriginFor;
+use sp_runtime::DispatchResult;
+
+use crate::{Config, DepositIdOf, AccountIdOf, BalanceOf};
+
+/// DepositChangeHandler is meant to be implemented on the runtime level to allow it to react to the event when a deposit changed
+pub trait DepositChangeHandler<T: Config> {
+    /// on_deposit_paid is called when a deposit was paid.
+    fn on_deposit_paid(
+        deposit: &DepositIdOf<T>,
+        account: &AccountIdOf<T>,
+        amount: BalanceOf<T>,
+    );
+
+    /// on_deposit_take_announcement is called when a deposit was withdrawn.
+    fn on_deposit_taken(
+        deposit: &DepositIdOf<T>,
+        account: &AccountIdOf<T>,
+        amount: BalanceOf<T>,
+    );
+}
+
+impl<T: Config> DepositChangeHandler<T> for () {
+    fn on_deposit_paid(_deposit: &DepositIdOf<T>, _account: &AccountIdOf<T>, _amount: BalanceOf<T>) {
+        // noop
+    }
+    fn on_deposit_taken(_deposit: &DepositIdOf<T>, _account: &AccountIdOf<T>, _amount: BalanceOf<T>) {
```

Comment:
Maybe would be good to either include total balance before and after, or the balance that was taken and the total balance remaining.